### PR TITLE
Materials library

### PR DESCRIPTION
**Pull request by julien-moreau**
Materials library:
- Improved water material, the sourceMesh parameter is not needed
- added readme

### CODE REVIEW

File: materialsLibrary/materials/water/readme.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Water material
```
Comment (deltakosh) [inline materialsLibrary/materials/water/readme.md]: You should add links to demo on PG

File: materialsLibrary/test/index.html [hunk 2]
```diff
@@ -144,7 +144,7 @@
 				simple.diffuseTexture.uScale = 5;
 				simple.diffuseTexture.vScale = 5;
 				
-				var water = new BABYLON.WaterMaterial("water", scene, currentMesh);
+				var water = new BABYLON.WaterMaterial("water", scene);
 				water.backFaceCulling = false;
 				water.enableRenderTargets(false);
```
Comment (deltakosh) [inline materialsLibrary/test/index.html]: Why?

Comment (julien-moreau) [inline materialsLibrary/test/index.html]: The water material no longer needs a source mesh since we can have the mesh reference in the isReady function

Comment (deltakosh) [inline materialsLibrary/test/index.html]: my why is about the enableRenderTargets. Why this function?

File: materialsLibrary/materials/water/readme.md (hunk 1 above)
Comment (julien-moreau) [inline materialsLibrary/materials/water/readme.md]: In the PG I still need to reference the source mesh in the constructor :(

File: materialsLibrary/test/index.html (hunk 2 above)
Comment (julien-moreau) [inline materialsLibrary/test/index.html]: Oh, the functon definition is:
    public enableRenderTargets(enable: boolean): void;
true to enable and false to disable.
It is enabled by default

File: materialsLibrary/materials/water/readme.md (hunk 1 above)
Comment (deltakosh) [inline materialsLibrary/materials/water/readme.md]: Should be good now

File: materialsLibrary/test/index.html (hunk 2 above)
Comment (deltakosh) [inline materialsLibrary/test/index.html]: Ok understood!

**Pull request merged**